frontend-react/src/components/EditableCompare.tsx

```diff
@@ -48,7 +49,7 @@ interface EditableCompareProps {
 export const EditableCompare = forwardRef(
     // allows for functions on components (useImperativeHandle)
     (
-        props: EditableCompareProps,
+        { jsonDiffMode, modified, onChange, original }: EditableCompareProps,
```

Comment:
Destructuring `props` here because using `props` as a dependency for hooks is bad practice since the reference will change if _any_ prop changes.  I've updated the dependencies accordingly below.